coop_point_of_sale/models/pos_order.py

```diff
@@ -0,0 +1,103 @@
+    week_day = fields.Char(
+        string="Day", compute="compute_week_day", store=True
+    )
+    cycle = fields.Char(string="Cycle", compute="compute_cycle", store=True)
+
+    @api.multi
+    @api.depends("date_order")
+    def compute_week_number(self):
+        for order in self:
+            if not order.date_order:
+                order.week_number = False
+            else:
+                weekA_date = fields.Date.from_string(
+                    self.env.ref("coop_shift.config_parameter_weekA").value
+                )
+                date_order = fields.Date.from_string(order.date_order)
+                week_number = 1 + (((date_order - weekA_date).days // 7) % 4)
+                if week_number == 1:
+                    order.week_number = "A"
+                elif week_number == 2:
+                    order.week_number = "B"
+                elif week_number == 3:
+                    order.week_number = "C"
+                elif week_number == 4:
+                    order.week_number = "D"
+
+    @api.multi
+    @api.depends("date_order")
+    def compute_week_day(self):
+        for order in self:
+            if order.date_order:
+                wd = order.date_order.weekday()
+                if wd == 0:
+                    order.week_day = _("Mon")
+                elif wd == 1:
+                    order.week_day = _("Tue")
+                elif wd == 2:
+                    order.week_day = _("Wes")
+                elif wd == 3:
+                    order.week_day = _("Thu")
+                elif wd == 4:
+                    order.week_day = _("Fri")
+                elif wd == 5:
+                    order.week_day = _("Sat")
+                elif wd == 6:
+                    order.week_day = _("Sun")
+
+    @api.multi
+    @api.depends("week_number", "week_day")
+    def compute_cycle(self):
+        for order in self:
+            order.cycle = "%s%s" % (order.week_number, order.week_day)
+
+    # Custom Section
+    @api.multi
+    def update_cycle_pos_order(self):
+
+        order_ids = self.ids
+        num_order_per_job = 20
+        splited_order_list = [
+            order_ids[i : i + num_order_per_job]
+            for i in range(0, len(order_ids), num_order_per_job)
+        ]
+        # Prepare session for job
+        # Create jobs
+        for order_list in splited_order_list:
+            self.update_cycle_pos_order_job("pos.order", order_list)
+        return True
+
+    @api.model
+    def _order_fields(self, ui_order):
+        order_fields = super(PosOrder, self)._order_fields(ui_order)
+        lines_value_lst = order_fields.get("lines", [])
+        updated_lines_lst = []
+        for line_values in lines_value_lst:
+            if line_values[-1].get("qty", 0):
+                updated_lines_lst.append(line_values)
+        order_fields["lines"] = updated_lines_lst
+        return order_fields
+
+    @job
+    @api.model
+    def update_cycle_pos_order_job(self, model_name, order_list):
```

Comment:
This function has the `@job` decorator but queue_job is not in the list of dependencies (not directly, at least. Maybe it's depended by coop_membership?)

Anyways, do we really need queue_jobs here?
Can't these fields be a standard computed field with depends?

When is `update_cycle_pos_order` even ran? I can't find any action.
At first glance it looks like you can remove both `update_cycle_pos_order` and `update_cycle_pos_order_job`, and the `queue_job` import